plugin/src/main/java/org/opensearch/sql/plugin/transport/TransportSQLService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.plugin.transport;
+
+import org.json.JSONObject;
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.action.ActionListener;
+import org.opensearch.commons.sql.action.TransportSQLQueryRequest;
+import org.opensearch.commons.sql.action.TransportSQLQueryResponse;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sql.common.antlr.SyntaxCheckException;
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.exception.ExpressionEvaluationException;
+import org.opensearch.sql.exception.QueryEngineException;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.protocol.response.QueryResult;
+import org.opensearch.sql.protocol.response.format.CsvResponseFormatter;
+import org.opensearch.sql.protocol.response.format.Format;
+import org.opensearch.sql.protocol.response.format.RawResponseFormatter;
+import org.opensearch.sql.protocol.response.format.ResponseFormatter;
+import org.opensearch.sql.protocol.response.format.SimpleJsonResponseFormatter;
+import org.opensearch.sql.sql.SQLService;
+import org.opensearch.sql.sql.domain.SQLQueryRequest;
+
+import java.util.Collections;
+
+import static org.opensearch.rest.RestStatus.BAD_REQUEST;
+import static org.opensearch.rest.RestStatus.SERVICE_UNAVAILABLE;
+import static org.opensearch.sql.legacy.plugin.RestSqlAction.QUERY_API_ENDPOINT;
+import static org.opensearch.sql.protocol.response.format.JsonResponseFormatter.Style.PRETTY;
+
+public class TransportSQLService {
```

Comment:
It seems TransportSQLService is a static class. Should we merge TransportSQLService into TransportSQLQueryAction? My previous comments maybe misleading.